torch_em/multi_gpu_training.py

```diff
@@ -33,37 +33,82 @@ def _create_data_loader(ds_callable, ds_kwargs, loader_kwargs, world_size, rank)
     return loader
 
 
+class DDP(DistributedDataParallel):
+    def __getattr__(self, name):
+        try:
+            return super().__getattr__(name)
+        except AttributeError:
+            return getattr(self.module, name)
+
+
 def _train_impl(
-    rank, world_size,
-    model_callable, model_kwargs,
-    train_dataset_callable, train_dataset_kwargs,
-    val_dataset_callable, val_dataset_kwargs,
-    loader_kwargs, iterations, **kwargs
+    rank,
+    world_size,
+    model_callable,
+    model_kwargs,
+    train_dataset_callable,
+    train_dataset_kwargs,
+    val_dataset_callable,
+    val_dataset_kwargs,
+    loader_kwargs,
+    iterations,
+    find_unused_parameters=False,
+    optimizer_callable=None,
+    optimizer_kwargs=None,
+    lr_scheduler_callable=None,
+    lr_scheduler_kwargs=None,
+    trainer_callable=None,
+    **kwargs
 ):
     assert "device" not in kwargs
     print(f"Running DDP on rank {rank}.")
     setup(rank, world_size)
 
     model = model_callable(**model_kwargs).to(rank)
-    ddp_model = DDP(model, device_ids=[rank])
+    ddp_model = DDP(model, device_ids=[rank], find_unused_parameters=find_unused_parameters)
+
+    if optimizer_callable is not None:
+        optimizer = optimizer_callable(model.parameters(), **optimizer_kwargs)
+        kwargs["optimizer"] = optimizer
+        if lr_scheduler_callable is not None:
+            lr_scheduler = lr_scheduler_callable(optimizer, **lr_scheduler_kwargs)
+            kwargs["lr_scheduler"] = lr_scheduler
 
     train_loader = _create_data_loader(train_dataset_callable, train_dataset_kwargs, loader_kwargs, world_size, rank)
     val_loader = _create_data_loader(val_dataset_callable, val_dataset_kwargs, loader_kwargs, world_size, rank)
 
-    trainer = torch_em.default_segmentation_trainer(
-        model=ddp_model, train_loader=train_loader, val_loader=val_loader,
-        device=rank, rank=rank, **kwargs
+    if trainer_callable is None:
+        trainer_callable = torch_em.default_segmentation_trainer
+
+    trainer = trainer_callable(
+        model=ddp_model,
+        train_loader=train_loader,
+        val_loader=val_loader,
+        device=rank,
+        rank=rank,
+        **kwargs
     )
     trainer.fit(iterations=iterations)
 
     cleanup()
 
 
 def train_multi_gpu(
-    model_callable, model_kwargs,
-    train_dataset_callable, train_dataset_kwargs,
-    val_dataset_callable, val_dataset_kwargs,
-    loader_kwargs, iterations, **kwargs
+    model_callable,
+    model_kwargs,
+    train_dataset_callable,
+    train_dataset_kwargs,
+    val_dataset_callable,
+    val_dataset_kwargs,
+    loader_kwargs,
+    iterations,
+    find_unused_parameters=False,
```

Comment:
From what I understand this is important to make the training work for more complex models. So should we set the default to `True`?

Reply:
Yep, you are correct. I've set the default value to True now. Thanks for the suggestion!